### ISSUE TITLE
Cannot update or delete VTL auth process from schema.graphql

### ISSUE THREAD

**Issue by ShinIchiTsuji** [aws-amplify/amplify-cli#5768]
**Describe the bug**
I want to edit the auth directive, but the output doesn't seem to accept any changes.

**Amplify CLI Version**
I tried at
amplify-cli@4.29.8
amplify-cli@4.30.0
amplify-cli@4.31.1
amplify-cli@4.32.0

**To Reproduce**
Edit schema.graphql

```
type Test
@model
@auth (rules: [
    {allow: groups, groups: ["admin"], operations: [create, update, delete]},
    {allow: private, operations: [read]}
  ])
{
  id: ID
  val: Int!
  str: String!
}
```
Run in terminal `amplify mock api`
The intended behavior of mutations and queries.

Edit and save schema.graphql (Add manager to permission group) leaving the terminal as it is
```
type Test
@model
@auth (rules: [
    {allow: groups, groups: ["admin", "manager"], operations: [create, update, delete]},
    {allow: private, operations: [read]}
  ])
{
  id: ID
  val: Int!
  str: String!
}
```
The following is displayed in the terminal

```
GraphQL Schema change detected. Reloading ...
GraphQL schema compiled successfully.
```

However, the manager cannot any mutation.

Notice appears in terminal when @auth directive is removed from Test model,
```
GraphQL Schema change detected. Reloading ...
The following types do not have'@auth' enabled. Consider using @auth with @model
         - Test
```
but authorization process has not changed from the initial state.

Creating a new model with a different name will result in the intended behavior.
```
type AnotherTest
@model
@auth (rules: [
    {allow: groups, groups: ["admin", "manager"], operations: [create, update, delete]},
    {allow: private, operations: [read]}
  ])
{
  id: ID
  val: Int!
  str: String!
}
```

**Expected behavior**
Is the authorization process updated?
The generated VTL file does not appear to be affected.

**Desktop (please complete the following information):**
 - OS: Windows
 - Node Version. v12.18.3

If this is the case, it seems like a big impact, but if it's my operational mistake, I'm very sorry.
I need help with the solution.

**Comment by akshbhu** [comment]
Hi @ShinIchiTsuji 

It seems you are checking the build folder inside your project folder. Can you check the resolvers folder , VTL template for createMutation will have the manager group added in the VTL. Resolver folder outside build is for ```amplify mock api``` and to debug VTL templates.

The build folder will get updated when you update the schema and use ```amplify api gql-compile``` or do an amplify push.

Let me know if that works for you?

**Comment by ShinIchiTsuji** [comment]
Hi akshbhu, thank you your reply.

Maybe it was my misunderstanding?
I run `amplify api gql-compile` first and then checked each part, but everything is working as intended.
It's a problem that has plagued me since last week, but as of this morning I haven't been able to reproduce it.
I'm very sorry.

I seem to have to re-verify after taking a short break...
If I can reproduce it accurately, I will consult again. Thank you.

**Comment by ShinIchiTsuji** [comment]
Hi @akshbhu
I'm terribly sorry for i do with restless...
But the VTL output still seems a bit strange.

Also, I didn't answer your question!
The location of the VTL file I'm referencing is `(app-name)\resolvers\*.vtl`

[schema.graphql.txt](https://github.com/aws-amplify/amplify-cli/files/5486344/schema.graphql.txt)
Run `amplify mock api` on the schema.graphql file I attached updates some files.At this time, the number of vtl files in the resolvers folder will be 117 or 54.
This number of vtl files switched between each other by overwriting schema.graphql without any changes.
Is this increase / decrease in the number of files normal?

Also, some files are not updated.

![スクリーンショット 2020-11-04 173004](https://user-images.githubusercontent.com/26863574/98088056-55646c00-1ec4-11eb-822d-1d7fa8795648.png)

Mutation.createCourse.req.vtl is old as shown in the attached image, so I comment out the @auth directive in Course and run the `amplify mock api`.
The terminal will give a notice @auth is not used in Course, but the file will not be updated and the `## [Start] Owner Authorization Checks **` item will still exist.

If I run `amplify api gql-compile` in this state, all vtl files in the build folder will be updated with the modification date and time, but the contents are the same as the Resolver folder outside the build folder.
The same is true for the AWS AppSync Console after ʻamplify push`.

I also tried starting the terminal with administrator privileges and operating from another terminal, but there was no change.

Will it be a clue to the solution?

**Comment by akshbhu** [comment]
Hi @ShinIchiTsuji 

1. No. of the VTL files depends on the directives present in your schema. For a simple case of @model directive , this will generate VTL templates for Queries and Mutations. You can find detailed documentation here : https://docs.amplify.aws/cli/graphql-transformer/model#generates

2. I guess you have updated the contents of the VTL templates highlighted above. If the contents are changes these resolvers will be counted as custom resolvers and ```amplify api gql-compile``` wont touch those templates as those will be your custom resolver templates and not generated by amplify and will be uploaded directly. Here is the documentation for custom resolvers : https://docs.amplify.aws/cli/graphql-transformer/resolvers.

if this a test project (not for production)what you can try is delete the resolvers present in the resolvers directory and try running ```amplify mock api```, it will create all new resolvers as per your schema and   after that if you do ```amplify api gql-compile``` ,this will generate resolvers in ```build/resolvers``` directory which will be similar to ```resolvers``` directory generated by mock.

Let me know if this answers your question.

**Comment by ShinIchiTsuji** [comment]
Hi @akshbhu

First, I could fix the problem by deleting all the vtl files in the resolver folder.
Thank you very much.

Apparently I was unintentionally creating a custom resolver to prevent vtl file generation from schema. I don't remember touching the vtl file, but it's probably the case.

(As an aside, I created a new folder in that location and moved the vtl file, but `amplify mock api` return `Failed to start API Mock endpoint Error: EISDIR: illegal operation on a directory, read` I returned it, so I moved to a completely different directory and remove new folder will work.)

Thank you for your patience with my poor English and explanations.
